src/Core/AdminConsole/OrganizationFeatures/Policies/PolicyValidators/AutomaticUserConfirmationPolicyValidator.cs

```diff
@@ -0,0 +1,131 @@
+/// </ul>
+/// </summary>
+public class AutomaticUserConfirmationPolicyValidator(
+    IOrganizationUserRepository organizationUserRepository,
+    IProviderUserRepository providerUserRepository,
+    IPolicyRepository policyRepository,
+    IOrganizationRepository organizationRepository,
+    TimeProvider timeProvider)
+    : IPolicyValidator, IPolicyValidationEvent
+{
+    public PolicyType Type => PolicyType.AutomaticUserConfirmation;
+
+    private const string _singleOrgPolicyNotEnabledErrorMessage =
+        "The Single organization policy must be enabled before enabling the Automatically confirm invited users policy.";
+
+    private const string _usersNotCompliantWithSingleOrgErrorMessage =
+        "All organization users must be compliant with the Single organization policy before enabling the Automatically confirm invited users policy. Please remove users who are members of multiple organizations.";
+
+    private const string _providerUsersExistErrorMessage =
+        "The organization has users with the Provider user type. Please remove provider users before enabling the Automatically confirm invited users policy.";
+
+    public IEnumerable<PolicyType> RequiredPolicies => [PolicyType.SingleOrg];
+
+    public async Task<string> ValidateAsync(PolicyUpdate policyUpdate, Policy? currentPolicy)
+    {
+        // Only perform validation when the policy is being enabled (previously disabled or null → now enabled)
+        if (policyUpdate is not { Enabled: true })
+        {
+            return string.Empty;
+        }
+
+        // If current policy is already enabled, no validation needed
+        if (currentPolicy is { Enabled: true })
+        {
+            return string.Empty;
+        }
+
+        return await ValidateEnablingPolicyAsync(policyUpdate.OrganizationId);
+    }
+
+    public async Task<string> ValidateAsync(SavePolicyModel savePolicyModel, Policy? currentPolicy) =>
+        await ValidateAsync(savePolicyModel.PolicyUpdate, currentPolicy);
+
+    public async Task OnSaveSideEffectsAsync(PolicyUpdate policyUpdate, Policy? currentPolicy)
+    {
+        var organization = await organizationRepository.GetByIdAsync(policyUpdate.OrganizationId);
+
+        if (organization is not null)
+        {
+            organization.UseAutomaticUserConfirmation = policyUpdate.Enabled;
+            organization.RevisionDate = timeProvider.GetUtcNow().UtcDateTime;
+            await organizationRepository.UpsertAsync(organization);
+        }
+    }
+
+    private async Task<string> ValidateEnablingPolicyAsync(Guid organizationId)
+    {
+        // Check 1: Validate Single Org policy is enabled and all users are compliant
+        var singleOrgValidationError = await ValidateSingleOrgPolicyComplianceAsync(organizationId);
+        if (!string.IsNullOrWhiteSpace(singleOrgValidationError))
+        {
+            return singleOrgValidationError;
+        }
+
+        // Check 2: Validate no provider users exist
```

Comment:
Non-blocker: consider removing these comments since the method should provide enough context, and the comments just add noise.